### PR TITLE
Adding stop() method.

### PR DESCRIPTION
**Pull request by mikeal**
It turns out that cleanly stoping and emitting end is a little tricky, this method helps. I've tested it in my app a bunch of times and it has never failed to end.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -25,6 +25,11 @@ function createRecordStream (media, opts) {
     rs.media = null
   }
 
+  rs.stop = function () {
+    rs.once('data', () => rs.emit('end'))
```
Comment (mafintosh) [inline index.js]: should be rs.push(null) instead of emit('end').

**Comment by mafintosh** [comment]
looks great except my comment! :)

**Comment by mikeal** [comment]
Fixed :)

**Comment by mafintosh** [comment]
2.1.0 \o/

**Pull request merged**